### PR TITLE
TierQuotaDefinitions are not deleted

### PR DESCRIPTION
**Pull request by Dannyps**
# Readiness checklist

-   [x] I added/updated unit tests.
-   [x] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

After an intensive round of tests, @JLSRKonk found that deleting a Tier did not delete the related TierQuotaDefinitions. Instead, it set their `TierId` to _ null_. This is not the expected behaviour, and this PR fixes that.
We also took the opportunity to change a number of other aspects about the application:
1.  The deletion of a TierQuotaDefinion is now done via its own repo method.
1. When a TierQuotaDefinition is deleted, the IntegrationEvent does not delete the related TierQuotas, because **they're now deleted via FK Cascade**.
1. When a TierQuotaDefinition is deleted, the IntegrationEvent recalculates MetricStatuses for all Identities with that TQD's Tier.
1. When a Tier is deleted, related TierQuotaDefinitions are deleted via FK Cascade.
1. When deleting a Tier from Quotas.Tiers, related Quotas.Identities are not subject to Cascade deletion. This was creating a double way for another entity, creating a circular cascade delete error.

### CODE REVIEW

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/TiersRepository .cs [hunk 1]
```diff
@@ -48,6 +50,15 @@ public async Task<TierQuotaDefinition> FindTierQuotaDefinition(string id, Cancel
...
 
+    public async Task RemoveTierQuotaDefinitionById(string id)
+    {
+        var deletedCount = await _tierQuotaDefinitions.Where(t => t.Id == id).ExecuteDeleteAsync();
+        if (deletedCount == 0)
+        {
+            throw new DomainException(GenericDomainErrors.NotFound("Tier Quota Definition"));
+        }
+    }
+
```
Comment (Dannyps) [inline Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/TiersRepository .cs:61]: I'm unsure about this Exception

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs [hunk 2]
```diff
@@ -5,7 +5,9 @@ namespace Backbone.Modules.Quotas.Domain.Aggregates.Identities;
 
 public class TierQuota : Quota
 {
-    private readonly TierQuotaDefinition _definition;
+#pragma warning disable IDE1006 // Naming Styles
+    internal readonly TierQuotaDefinition _definition;
```
[the remark this answers is not in the document]
Comment (Dannyps) [reply Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs:9]: This was necessary to set the cascade behaviour on the `TierQuotaEntityTypeConfiguration.cs` file.

[the remark this answers is not in the document]
Comment (tnotheis) [reply Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs:9]: As I said: 

> Afaik they can work with private fields as well. Did you try it or was it just an assumption?

[the remark this answers is not in the document]
Comment (tnotheis) [reply Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs:9]: Soo... does the following not work?

```c#
builder.HasOne("_definition").WithMany().OnDelete(DeleteBehavior.Cascade);
```

[the remark this answers is not in the document]
Comment (Dannyps) [reply Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs:9]: Sorry. Yes, setting it to private throws a build error:

`Error	CS0122	'TierQuota._definition' is inaccessible due to its protection level	Quotas.Infrastructure	enmeshed\backbone\Modules\Quotas\src\Quotas.Infrastructure\Persistence\Database\EntityConfigurations\TierQuotaEntityTypeConfiguration.cs`

[the remark this answers is not in the document]
Comment (Dannyps) [reply Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/TierQuota.cs]: Aye, that does work!

**Review by tnotheis — APPROVED** [review]

**Pull request merged**